From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ruViolence <78062896+ruViolence@users.noreply.github.com>
Date: Wed, 8 Jun 2022 10:49:06 +0500
Subject: [PATCH] feat: configurable time update period

Allows breaking client-side tickrate meters and also slightly reduce the network load.

diff --git a/src/main/java/com/github/ruviolence/reaper/ReaperConfig.java b/src/main/java/com/github/ruviolence/reaper/ReaperConfig.java
index b634d2375f167e20b495e95fa25977c3f3a8d52c..50eec3e6050612f5e2dc430d76cddc6fa00992e1 100644
--- a/src/main/java/com/github/ruviolence/reaper/ReaperConfig.java
+++ b/src/main/java/com/github/ruviolence/reaper/ReaperConfig.java
@@ -249,4 +249,9 @@ public class ReaperConfig {
     private static void alternativeChatComponentParser() {
         alternativeChatComponentParser = getBoolean("alternative-chat-component-parser", false);
     }
+
+    public static int timeUpdatePeriod;
+    private static void timeUpdatePeriod() {
+        timeUpdatePeriod = getInt("time-update-period", 20);
+    }
 }
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index fa4e353e83a0bcd55077d11cac48dc845bd3a176..58dcbba12e4a364c16b6e775b62bc2d73d43b46d 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -124,6 +124,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
     // Spigot start
     public final SlackActivityAccountant slackActivityAccountant = new SlackActivityAccountant();
     // Spigot end
+    public int timeUpdateTicks; // Reaper - Custom time update period
     // Reaper start
     boolean hasEntityMoveEvent;
     boolean hasEntityTickEvent;
@@ -884,7 +885,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
 
         MinecraftTimings.timeUpdateTimer.startTiming(); // Spigot
         // Send time updates to everyone, it will get the right time from the world the player is in.
-        if (this.ticks % 20 == 0) {
+        if (++this.timeUpdateTicks > com.github.ruviolence.reaper.ReaperConfig.timeUpdatePeriod) { this.timeUpdateTicks = 0; // Reaper - Custom time update period
             for (int i = 0; i < this.getPlayerList().players.size(); ++i) {
                 EntityPlayer entityplayer = (EntityPlayer) this.getPlayerList().players.get(i);
                 entityplayer.playerConnection.sendPacket(new PacketPlayOutUpdateTime(entityplayer.world.getTime(), entityplayer.getPlayerTime(), entityplayer.world.getGameRules().getBoolean("doDaylightCycle"))); // Add support for per player time
